package.json

```diff
@@ -1,14 +1,27 @@
 {
   "name": "@questdb/nodejs-client",
-  "version": "3.0.0",
+  "version": "4.0.0",
   "description": "QuestDB Node.js Client",
-  "main": "index.js",
-  "types": "types/index.d.ts",
   "scripts": {
-    "test": "jest",
-    "eslint": "npx eslint src/**",
-    "docs": "jsdoc src/sender.js src/options.js README.md -d docs",
-    "types": "npx -p typescript tsc --skipLibCheck --declaration --allowJs --emitDeclarationOnly --outDir types"
+    "test": "vitest",
```

Comment:
No, it won't. Only in local which is good, on CI Vitest auto-detects that is run inside a ci environment and runs in "run" mode

I'm curious to see why it fails in your machine, look at CI on my fork: https://github.com/semoal/nodejs-questdb-client/actions also on my machine passes good. Review that you're using at least Node 20

Reply:
Yeah, just tried with Node 18.17.1 and crashes with your errors, following community standard:
!(https://raw.githubusercontent.com/nodejs/Release/main/schedule.svg?sanitize=true)[https://raw.githubusercontent.com/nodejs/Release/main/schedule.svg?sanitize=true]
It is on maintenance mode already

Reply:
does this mean that the library would not work on node18?

Reply:
let's add node 18 to the tests as well. it is still active and used

Reply:
3.0.0 will work for Node 18 and less and should receive maintenance until April 2025 at least, to follow Node conventions. 
4.0.0 will work for Node 20 and above, we must follow industry standard, not what people use.

Look several examples:
> Fastify v5 will only support Node.js v20+. If you are using an older version of Node.js, you will need to upgrade to a newer version to use Fastify v5.

Reply:
what's the point of the library that uses standards when nobody uses it?

Reply:
I don't understand your discussion here, this is why is a semver major because includes breaking changes, this is a normal lifecycle in a library powered or nearly ligated to Node.

Reply:
I think we should make backward-compatible PR to implement CJS and ESM compilation without adding new stuff.  This way, everyone benefits. Later in the new PRs, we can help them introduce small changes like adding undici for greater performance. 

I am pro this PR but not if it makes a big change so the rest of the users don't benefit.

As you can see, the core team seems busy, and this PR is quite a challenge to review and even more so to approve. 
<img width="751" alt="image" src="https://github.com/user-attachments/assets/b8bff254-d348-433e-94ae-baa787ed5f6f" />